### DISCUSSION TITLE
MSSQL Prisma: Cannot read property 'findMany' of undefined"

### DISCUSSION THREAD

**Discussion by BJladika** [prisma/prisma#7475] [Q&A] [answered]
Pretty new to prisma. In short followed Apollo-Server + Prisma example: https://github.com/prisma/prisma-examples/tree/latest/javascript/graphql
But with replacing SQLite database with MSSQL database on localhost

Have test table people_List with a few columns like 
`id_People, Name, Surname`

prisma introspect generated

replaced code at src/Schema.js
for smth like: 
```
const Query = objectType({
  name: 'Query',
  definition(t) {
    t.nonNull.list.nonNull.field('students', {
      type: 'people_List',
      resolve: (_parent, _args, context) => {
        return context.prisma.people_list.findMany()
      },
    })
  },
})

const Student = objectType({
  name: 'people_List',
  definition(t) {
    t.nonNull.int('id_People')
    t.string('Name')
    t.string('Surname')
  },
})
```

and it made schema.graphql like:
```
type Query {
  students: [people_List!]!
}

type people_List {
  Name: String
  Surname: String
  id_People: Int!
}
```

changed schema.prisma to
```
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["microsoftSqlServer"]
}

datasource db {
  provider = "sqlserver"
  url      = env("DATABASE_URL")
}

model people_List {
  id_People           Int       @id @default(autoincrement())
  Surname             String    @db.VarChar(30)
  Name                String    @db.VarChar(30)
}
```

launched Playground tried to make query and receiving "Cannot read property 'findMany' of undefined" error.
Where did I go wrong?

**Comment by janpio** [comment, marked as answer]
`prisma.people_list` is not a valid model in your Prisma Client, as the model is called `people_List`. 

If you replace `        return context.prisma.people_list.findMany()` with `        return context.prisma.people_List.findMany()` it should work.

**Reply by BJladika** [reply]
Well, I just read all these examples and it kinda implied that I must write all these thingie with lowercase like ye prisma.user instead of prisma.User

Also now I've encountered some weird error
```Invalid `prisma.people_List.findMany()` invocation:\n\n\n  Error occurred during query execution:\nConnectorError(ConnectorError { user_facing_error: None, kind: QueryError(Utf8) })```

**Reply by janpio** [reply]
The lowercasing only is for the first letter of the model name, as that is what the Typescript API of Prisma Client generates. If you use a editor with code completion, that should show you the list of available models under `prisma` for you to use. (VSCode with the Prisma extension is quite nice in many other regards as well)

For that error, that is weird. I suggest you open a new issue or discussion and include some more information.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
